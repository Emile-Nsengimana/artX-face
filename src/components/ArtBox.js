import React from 'react';

// eslint-disable-next-line react/prefer-stateless-function
class ArtBox extends React.Component {
  render() {
    return (
      <>
        {this.props.images.length !== 0
          ? this.props.images.map((img) => (
            <div key={img}>
              <img src={img} alt="" />
              <p>It is a long established fact that a reader will be distracted by the readable content of a page when looking at its layout. The point of using Lorem Ipsum is that it has a more-or-less normal distribution of letters, as opposed to using 'Content here, content here', making it look like readable English</p>
            </div>

          ))
          : <h3 className="not-found">Not found</h3>}
      </>
    );
  }
}

export default ArtBox;
