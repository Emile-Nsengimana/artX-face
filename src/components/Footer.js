import React from 'react';
import mastercard from '../img/mastercard.jpg';
import mtn from '../img/mtn.png';
import airtelTigo from '../img/airtel-tigo.jpg';

const Footer = () => (
  <div className="last">
    <a href="#" alt="">Contact us</a>
    <a href="#" alt="">Privacy policy</a>
    <a href="#" alt="">Terms of use</a>
    <div className="card">
      <img src={mastercard} alt="" />
      <img src={mtn} alt="" />
      <img src={airtelTigo} alt="" />
    </div>
    <h6>&copy;2020 ARTX. ALL RIGHTS RESERVED.</h6>
  </div>
);

export default Footer;
