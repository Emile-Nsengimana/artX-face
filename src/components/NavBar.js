import React from 'react';
import { Link } from 'react-router-dom';
import Logo from '../img/logo.svg';

const NavBar = () => (
  <div className="navbar">
    <div className="logo">
      <img src={Logo} alt="" />
    </div>
    <div className="left-menu">
      <Link to="/">HOME</Link>
      <Link to="/gallery">GALLERY</Link>
      <Link to="/about">ABOUT US</Link>
    </div>
    <div className="right-menu">
      <button type="button">Login</button>
      <button type="button" className="btn-round">Get Started</button>
    </div>
  </div>
);

export default NavBar;
