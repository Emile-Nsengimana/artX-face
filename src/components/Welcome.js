import React from 'react';
import BackgroundImg from '../img/home.jpg';
import Down from '../img/svg/down-arrow.svg';

const Welcome = () => (
  <>
    <div className="box-title">
      <h1 className="title-primary">
        <span className="title-primary-top">artx</span>
        <span className="title-primary-sub">your art store</span>
      </h1>
    </div>
    <div className="down-arrow">
      <a href="/"><img src={Down} alt="" /></a>
    </div>
  </>
);

export default Welcome;
