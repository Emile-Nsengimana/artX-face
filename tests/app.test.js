import React from 'react';
import renderer from 'react-test-renderer';
import NavBar from '../src/components/NavBar';
import Welcome from '../src/components/Welcome';
import ArtBox from '../src/components/ArtBox';
import Footer from '../src/components/Footer';

describe('Landing page components', () => {
  test('navbar component', () => {
    const Component = renderer.create(<NavBar />).toJSON();
    expect(Component).toMatchSnapshot();
  });

  test('welcome message component', () => {
    const MenuBar = renderer.create(<Welcome />).toJSON();
    expect(MenuBar).toMatchSnapshot();
  });

  test('art box component without images', () => {
    const images = [];
    const Component = renderer.create(<ArtBox images={images} />).toJSON();
    expect(Component).toMatchSnapshot();
  });

  test('art box component with images', () => {
    const images = [
      1, 2,
    ];
    const Component = renderer.create(<ArtBox images={images} />).toJSON();
    expect(Component).toMatchSnapshot();
  });

  test('footer component', () => {
    const Component = renderer.create(<Footer />).toJSON();
    expect(Component).toMatchSnapshot();
  });
});
