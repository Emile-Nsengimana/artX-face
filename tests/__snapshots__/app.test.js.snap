// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a h1 welcome message 1`] = `
<div
  className="container"
>
  <div
    className="menu"
  >
    <div
      className="navbar"
    >
      <div
        className="logo"
      >
        <img
          alt=""
          src="test-file-stub"
        />
      </div>
      <div
        className="left-menu"
      >
        <a
          href="/"
          onClick={[Function]}
        >
          HOME
        </a>
        <a
          href="/gallery"
          onClick={[Function]}
        >
          GALLERY
        </a>
        <a
          href="/about"
          onClick={[Function]}
        >
          ABOUT US
        </a>
      </div>
      <div
        className="right-menu"
      >
        <button
          type="button"
        >
          Login
        </button>
        <button
          className="btn-round"
          type="button"
        >
          Get Started
        </button>
      </div>
    </div>
  </div>
  <div
    className="display"
  >
    <img
      alt=""
      className="slide"
      src="test-file-stub"
    />
  </div>
  <div
    className="content"
  />
  <div
    className="footer"
  />
</div>
`;
